pallets/automation-time/src/types.rs

```diff
@@ -35,6 +37,21 @@ pub enum Action<AccountId, Balance, CurrencyId> {
 	},
 }
 
+impl<AccountId, Balance, CurrencyId> Action<AccountId, Balance, CurrencyId> {
+	pub fn execution_weight<T: Config>(&self) -> Weight {
```

Comment:
out of curiosity did it not work to just use the `WeightInfo` trait?

Reply:
It did not because of `Action::XCMP { .. } => T::DbWeight::get()` (T::DbWeight).

Also, didn't love that I had to import `crate::BalanceOf` from lib for https://github.com/OAK-Foundation/OAK-blockchain/pull/257/files#diff-62dc10adefef24a3ca74065c929abab1f72492b3bdd47293d2e7d92c4d340a26R221 but I'm not sure how to since the ScheduledTask struct doesn't have a Balance type this seemed like the most ergonomic way to express define the Task type expected by this method.

Reply:
you just need to add a generic to function to avoid this. Just like you did for the`execution_weight`

Reply:
I made the change you suggested here: https://github.com/OAK-Foundation/OAK-blockchain/pull/257/commits/64151e01af23168baaabae73f8fb783c7bcf3e98
I have some long term concern about how clunky and hard to read all of these generics are making the code. It's unclear to me what the value we're gaining by making these structs so generic. I expect the clunkiness and poor readability is only going to get worse as we write more complex code with this pattern.
But, that's a problem for a different pr!